cmd/common/signer/signer.go

```diff
@@ -94,10 +95,18 @@ func validateEnrollmentCertificate(b []byte) error {
 
 func (si *Signer) Sign(msg []byte) ([]byte, error) {
 	digest := util.ComputeSHA256(msg)
-	return signECDSA(si.key, digest)
+	switch key := si.key.(type) {
+	// Fabric only supports ECDSA at the moment.
+	case *ecdsa.PrivateKey:
+		return signECDSA(si.key.(*ecdsa.PrivateKey), digest)
+	case ed25519.PrivateKey:
+		return ed25519.Sign(si.key.(ed25519.PrivateKey), digest), nil
```

Comment:
Is this correct? The godoc suggests that, while [ecdsa sign takes a hash/digest](https://pkg.go.dev/crypto/ecdsa@go1.18.4#Sign), the [ed25519 sign required the entire message bytes](https://pkg.go.dev/crypto/ed25519@go1.18.4#Sign). There is more detail in the PrivateKey Sign method doc:

https://pkg.go.dev/crypto/ed25519@go1.18.4#PrivateKey.Sign

Reply:
I am evaluating a possible solution, as I believe these requirements will lead to changes in the Signer interface

https://github.com/johannww/fabric-1/blob/9fdc4f0c3ac92a7c95ab0099ac6535a1fc4410f0/bccsp/signer/signer.go#L76-L78

So functions like the one below can be called:

https://github.com/johannww/fabric-1/blob/9fdc4f0c3ac92a7c95ab0099ac6535a1fc4410f0/msp/identities.go#L255-L278

Reply:
I think it is fixed. Plus, I forced-pushed some commits because they were not signed off.